### ISSUE TITLE
unexpected compiler error: attempt to expose non-public symbol ''

### ISSUE THREAD

**Issue by Gubaer** [ballerina-platform/ballerina-lang#35433]
**Description:**
`bal`compiler reports about a non-public symbol ''.

**Steps to reproduce:**
Build the following ballerina fragment: 
```ballerina
public class A {
    public function increment(0 | 1 value) returns int {
        return value + 1;
    }
}

public type ZeroOrOne 0 | 1;

public class B {
    public function increment(ZeroOrOne value) returns int {
        return value + 1;
    }
}

public function main() returns error? {
    var a = new A();
    var result = a.increment(0);
}
```
The compiler reports the following error:

```
Compiling source
	test.bal
ERROR [test.bal:(2:5,4:6)] attempt to expose non-public symbol ''
WARNING [test.bal:(17:5,17:33)] unused variable 'result'
error: compilation contains errors
```
This is probably related to the type `0 | 1`  on line 2.

**Affected Versions:**
Ballerina 2201.0.1 (Swan Lake)
Language specification 2022R1
Update Tool 1.3.9

**OS, DB, other environment details and versions:**

Linux integrationbox-02 5.13.0-35-generic #40~20.04.1-Ubuntu SMP Mon Mar 7 09:18:32 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

openjdk 17.0.2 2022-01-18
OpenJDK Runtime Environment (build 17.0.2+8-Ubuntu-120.04)
OpenJDK 64-Bit Server VM (build 17.0.2+8-Ubuntu-120.04, mixed mode, sharing)

**Comment by lochana-chathura** [comment]
Related issue #34839

**Comment by rdulmina** [comment]
Fixed with  #36959
